pkg/client/client.go

```diff
@@ -126,3 +151,30 @@ func parseRevertReason(bz []byte) (string, error) {
 	size.SetBytes(bz[36:68])
 	return string(bz[68 : 68+size.Int64()]), nil
 }
+
+type Result struct {
+	Type         *string  `json:"type"`
+	From         *string  `json:"from"`
+	To           *string  `json:"to"`
+	Value        *string  `json:"value"`
+	Gas          *string  `json:"gas"`
+	GasUsed      *string  `json:"gasUsed"`
+	Input        *string  `json:"input"`
+	Output       *string  `json:"output"`
+	Error        *string  `json:"error"`
+	RevertReason *string  `json:"revertReason"`
+	Calls        []Result `json:"calls"`
+}
+
+func searchToAndReason(to string, calls []Result) (string, string) {
+	for _, call := range calls {
+		if call.To != nil {
+			to = *call.To
+		}
+		if call.RevertReason != nil {
+			return to, *call.RevertReason
+		}
```

Comment:
Can there be a case where `call.RevertReason != nil` but `call.To == nil`?
If any, when does it happen?

Reply:
I think the following function will make the code more natural.

```
func searchToAndReason(result Result) (string, string, error) {
    if result.RevertReason != nil {
        return *result.To, *result.RevertReason, nil
    }
    for _, call := range result.Calls {
        to, reason, err := searchToAndReason(call)
        if err == nil {
            return to, reason, nil
        }
    }
    return "", "", fmt.Errorf("revert reason not found")
}
```

Reply:
Fixed search reason logic
https://github.com/datachainlab/ethereum-ibc-relay-chain/pull/10/commits/53a99c31942d630704e8abb75c5d2899fe179aec